### PR TITLE
Add dynamic transaction fee in sending Lisk tokens - Closes #2910

### LINKED ISSUE #2910: Add dynamic transaction fee in sending Lisk tokens
### Description

As a user, i would like to select the transaction fee (High/Low/Custom) instead of static fee whenever i transfer Lisk tokens from my accounts to others account. 

Transaction fee will be calculated by the lisk-service and retrieved by lisk-desktop based on user selecting the priority (High/Low/Medium/Custom)

### Validation details

- Input valid address.
- Input valid amount.
- The amount must be higher than 0.01 LSK
- The amount + fee must be less than balance.
- Input custom message.
- Choose low, medium or high priority. Or Input a custom fee

For 'Send LSK' user journey refer the below UI design

[Step 1](https://www.figma.com/proto/yy39tEdAaYoVjl6LaqgpyH/Prototype?node-id=131%3A446&viewport=-636%2C-268%2C0.13392853736877441&scaling=min-zoom) -->[Step 2](https://www.figma.com/proto/yy39tEdAaYoVjl6LaqgpyH/Prototype?node-id=1%3A3099&viewport=-636%2C-268%2C0.13392853736877441&scaling=min-zoom)-->[Step 3](https://www.figma.com/proto/yy39tEdAaYoVjl6LaqgpyH/Prototype?node-id=1%3A3607&viewport=-636%2C-268%2C0.13392853736877441&scaling=min-zoom)

### Acceptance Criteria
- Send Lisk form should show Low, Medium and High priority buttons. Users can select one of this option. The default option should be LOW
- Display the dynamic transaction fee amount based on the user selection
- Send Lisk form should provide users with a custom fee field. Base fee should be displayed as part of tool tip or validation message
- Show custom fee field if the Lisk Service API is not available.  Low/High fee buttons should be disabled
- Remove second signature sign-in from this page
- Display the dynamic transaction fee in the transaction summary page
- User should have an option to edit the transaction details before confirming the transaction
- View transaction success or failure result.

### Additional Information 
- Transaction fee must be calculated based on the transactions size according to 
recommendations of the corresponding [LIP](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0016.md)

### PR DESCRIPTION
**Pull request by UsamaHameed**
### What was the problem?
This PR partially resolves #2910 

### How was it solved?
Use the new generic dynamic fee component.

### How was it tested?
WIP

### CODE REVIEW

File: src/components/shared/dynamicFee/index.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+  return (
+    <div className={`${styles.dynamicFeeWrapper} ${styles.fieldGroup} processing-speed`}>
+      <div className={`${styles.col}`}>
+        <span className={`${styles.fieldLabel}`}>
+          {t('Priority')}
+          <Tooltip>
+            <p className={styles.tooltipText}>
+              {
+                t('Lorem Ipsum...')
```
Comment (reyraa) [inline src/components/shared/dynamicFee/index.js:87]: When the network is busy, transactions with higher priority get processed sooner.

File: src/components/screens/send/form/formBtc.js [hunk 2]
```diff
@@ -1,23 +1,18 @@
...
-  formatAmountBasedOnLocale,
-} from '../../../../utils/formattedNumber';
-import { fromRawLsk, toRawLsk } from '../../../../utils/lsk';
+import { toRawLsk } from '../../../../utils/lsk';
 import FormBase from './formBase';
-import Selector from '../../../toolbox/selector/selector';
-import Spinner from '../../../toolbox/spinner';
-import Tooltip from '../../../toolbox/tooltip/tooltip';
-import styles from './form.css';
+import DynamicFee from '../../../shared/dynamicFee';
```
Comment (reyraa) [inline src/components/screens/send/form/formBtc.js:4]: Please rename to `Priority`.

File: src/components/screens/send/form/formBtc.js [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
+const txType = 'transfer';
+
 const FormBtc = (props) => {
   const {
-    t, token, getInitialValue, account,
+    token, getInitialValue, account,
   } = props;
-  const txType = 'transfer';
 
   const [processingSpeed, selectProcessingSpeed, feeOptions] = useProcessingSpeed(token);
```
Comment (reyraa) [inline src/components/screens/send/form/formBtc.js:17]: Please change to
```
const [priority, setPriority, priorities] = usePriority(token);
```

File: src/components/screens/send/form/formBtc.test.js [hunk 4]
```diff
@@ -47,7 +45,9 @@ const dynamicFeeFactor = 100;
 getDynamicBaseFees.mockResolvedValue(dynamicBaseFees);
 getDynamicFee.mockImplementation((params) => {
   const selectedProcessingSpeed = params.dynamicFeePerByte.selectedIndex;
-  const fees = Object.values(dynamicBaseFees)[selectedProcessingSpeed] * dynamicFeeFactor;
+  const fees = fromRawLsk(
+    Object.values(dynamicBaseFees)[selectedProcessingSpeed] * dynamicFeeFactor,
```
Comment (reyraa) [inline src/components/screens/send/form/formBtc.test.js:49]: 1. `selectedPriority`
2. `baseFeeFactor` ?

File: src/components/screens/send/form/formBtc.test.js [hunk 5]
```diff
@@ -118,7 +114,7 @@ describe('FormBtc', () => {
       act(() => { jest.runAllTimers(); });
       wrapper.update();
       await flushPromises();
-      expect(wrapper.find('.amount input').prop('value')).toEqual(fromRawLsk(balance - dynamicBaseFees.Low * dynamicFeeFactor));
+      expect(wrapper.find('.amount input').prop('value')).toEqual(fromRawLsk(balance - (dynamicBaseFees.Low * dynamicFeeFactor)));
```
Comment (reyraa) [inline src/components/screens/send/form/formBtc.test.js:117]: `baseFees`

File: src/components/shared/dynamicFee/dynamicFee.css [hunk 6]
```diff
@@ -0,0 +1,85 @@
+@import '../../../app/mixins.css';
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.css:1]: Accordingly the css file should also be renamed to `priority.css`.

File: src/components/shared/dynamicFee/dynamicFee.css [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+.fieldGroup {
+  align-items: flex-start;
+  display: flex;
+  flex-direction: column;
+  margin-bottom: 16px;
+  padding: 0 var(--horizontal-padding-l);
+
+  &.dynamicFeeWrapper {
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.css:10]: `.feeWrapper`

File: src/components/shared/dynamicFee/dynamicFee.css [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  display: flex;
+  margin-bottom: 8px;
+
+  & .loading {
+    color: var(--color-ultramarine-blue);
+    margin-left: 5px;
+  }
+}
+
+.dynamicFeeWrapper {
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.css:30]: `.feeWrapper`

File: src/components/shared/dynamicFee/dynamicFee.css [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+      margin-bottom: 14px;
+    }
+
+    & > .prioritySelector {
+      display: flex;
+      flex-direction: row;
+      justify-content: space-between;
+      margin-bottom: 4px;
+
+      & > .feePriority {
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.css:52]: Not sure if this name makes sense. Fee and priority are different identities. I think it's meant to be `.priorityTitle`

File: src/components/shared/dynamicFee/dynamicFee.css [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        font-family: var(--content-font);
+        font-size: 14px;
+        line-height: 18px;
+        border: 1px solid var(--color-periwinkle-blue);
+        border-radius: 18px;
+        margin-right: 5px;
+        padding: 5px var(--horizontal-padding-m);
+        outline: none;
+
+        &.feePrioritySelected {
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.css:64]: Same here. it deserves a more descriptive/accurate name.

File: src/components/shared/dynamicFee/dynamicFee.test.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+// import { act } from 'react-dom/test-utils';
+import { mount } from 'enzyme';
+import { tokenMap } from '../../../constants/tokens';
+import DynamicFee from '.';
+// import accounts from '../../../../../test/constants/accounts';
+// import flushPromises from '../../../../../test/unit-test-utils/flushPromises';
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.test.js:7]: Please remove if not needed.

File: src/components/shared/dynamicFee/dynamicFee.test.js [hunk 12]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+// import { act } from 'react-dom/test-utils';
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.test.js:2]: Please rename if not needed.

File: src/components/shared/dynamicFee/dynamicFee.test.js [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+const baseFees = {
+  Low: 0.1,
+  Medium: 0.2,
+  High: 0.3,
+};
+
+describe('DynamicFee', () => {
+  let wrapper;
+  // let selectedPriority;
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.test.js:17]: Same here.

File: src/components/shared/dynamicFee/dynamicFee.test.js [hunk 14]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+// import { act } from 'react-dom/test-utils';
+import { mount } from 'enzyme';
+import { tokenMap } from '../../../constants/tokens';
+import DynamicFee from '.';
```
Comment (reyraa) [inline src/components/shared/dynamicFee/dynamicFee.test.js:5]: `Priority`

File: src/components/shared/dynamicFee/index.js [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+  const onClickCustomEdit = (e) => {
+    e.preventDefault();
+    setShowEditIcon(false);
+  };
+
+  const tokenRelevantPriorities = useMemo(() =>
+    priorities.filter((_, index) =>
+      index !== CUSTOM_FEE_INDEX
+      || (index === CUSTOM_FEE_INDEX && token === tokenMap.LSK.key)),
+  [token, priorities]);
```
Comment (reyraa) [inline src/components/shared/dynamicFee/index.js:64]: If you define this outside the component, you can reinstate the max-statement eslint rule.

File: src/components/shared/dynamicFee/index.js [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+  const isLoading = priorities[0].value === 0;
+
+  const getFeeStatus = () => {
+    if (customFee) {
+      return customFee;
+    }
+    return !fee.error
+      ? `${formatAmountBasedOnLocale({ value: fee.value })} ${token}`
+      : fee.feedback;
+  };
```
Comment (reyraa) [inline src/components/shared/dynamicFee/index.js:75]: This can move outside the component.

File: src/components/shared/dynamicFee/index.js [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+
+  const inputValue = !isLoading && (customFee === 'undefined' ? fee.value : customFee);
+
+  return (
+    <div className={`${styles.dynamicFeeWrapper} ${styles.fieldGroup} processing-speed`}>
```
Comment (reyraa) [inline src/components/shared/dynamicFee/index.js:80]: Please rename every variable with `dynamic` or `processing-speed` according the previous comments.

File: src/components/shared/dynamicFee/index.js [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+          ))}
+        </div>
+      </div>
+      <div className={`${styles.col} fee-container`}>
+        <span className={`${styles.fieldLabel}`}>
+          {t('Transaction fee')}
+          <Tooltip>
+            <p className={styles.tooltipText}>
+              {
+                t('Lorem Ipsum...')
```
Comment (UsamaHameed) [inline src/components/shared/dynamicFee/index.js:111]: @reyraa We need tooltip text here as well.

Comment (reyraa) [reply src/components/shared/dynamicFee/index.js:111]: You can choose a high, medium, or low transaction priority, each requiring a corresponding transaction fee. Or you can pay any desired fee of no less than [minFee] [LSK/BTC]. If you don't why, we recommend you choose one of the said options instead.

**Review by reyraa — CHANGES_REQUESTED** [review]

**Review by reyraa — APPROVED** [review]
Great job Usama. Thanks for resolving all the issues.

**Pull request merged**